toolsrc/src/vcpkg/binarycaching.cpp

```diff
@@ -244,12 +245,541 @@ namespace
+                            paths.get_triplet_info(action.abi_info.value_or_exit(VCPKG_LINE_INFO)),
+                            "\nFeatures: ",
+                            Strings::join(", ", action.feature_list),
+                            "\nDependencies:\n");
+
+        for (auto&& dep : action.package_dependencies)
+        {
+            Strings::append(description, "    ", dep.name(), '\n');
+        }
+        XmlSerializer xml;
+        xml.open_tag("package").line_break();
+        xml.open_tag("metadata").line_break();
+        xml.simple_tag("id", ref.id).line_break();
+        xml.simple_tag("version", ref.version).line_break();
+        if (!scf.core_paragraph->homepage.empty()) xml.simple_tag("projectUrl", scf.core_paragraph->homepage);
+        xml.simple_tag("authors", "vcpkg").line_break();
+        xml.simple_tag("description", description).line_break();
+        xml.open_tag("packageTypes");
+        xml.start_complex_open_tag("packageType").text_attr("name", "vcpkg").finish_self_closing_complex_tag();
+        xml.close_tag("packageTypes").line_break();
+        xml.close_tag("metadata").line_break();
+        xml.open_tag("files");
+        xml.start_complex_open_tag("file")
+            .text_attr("src", (paths.package_dir(spec) / fs::u8path("**")).u8string())
+            .text_attr("target", "")
+            .finish_self_closing_complex_tag();
+        xml.close_tag("files").line_break();
+        xml.close_tag("package").line_break();
+        paths.get_filesystem().write_contents(nuspec_path, xml.buf, VCPKG_LINE_INFO);
+    }
+
+    struct NugetBinaryProvider : IBinaryProvider
+    {
+        NugetBinaryProvider(std::vector<std::string>&& read_sources,
+                            std::vector<std::string>&& write_sources,
+                            std::vector<fs::path>&& read_configs,
+                            std::vector<fs::path>&& write_configs,
+                            bool interactive)
+            : m_read_sources(std::move(read_sources))
+            , m_write_sources(std::move(write_sources))
+            , m_read_configs(std::move(read_configs))
+            , m_write_configs(std::move(write_configs))
+            , m_interactive(interactive)
+        {
+        }
+        ~NugetBinaryProvider() = default;
+        void prefetch(const VcpkgPaths& paths, const Dependencies::ActionPlan& plan) override
+        {
+            if (m_read_sources.empty() && m_read_configs.empty()) return;
+            if (plan.install_actions.empty()) return;
+
+            auto& fs = paths.get_filesystem();
+
+            std::vector<std::pair<PackageSpec, NugetReference>> nuget_refs;
+
+            for (auto&& action : plan.install_actions)
+            {
+                auto& spec = action.spec;
+                fs.remove_all(paths.package_dir(spec), VCPKG_LINE_INFO);
+
+                NugetReference nuget_ref(action);
+
+                nuget_refs.emplace_back(spec, std::move(nuget_ref));

```

Comment:
```suggestion
                nuget_refs.emplace_back(spec, NugetReference{action});

```

Super nitpick, up to you